platforms/common/uORB/uORBManager.hpp

```diff
@@ -446,15 +446,22 @@ class Manager
 
 	static void queueCallback(class SubscriptionCallback *sub, int idx)
 	{
-		_Instance->g_sem_pool.cb_lock(idx);
-		_Instance->g_sem_pool.cb_set(idx, sub);
-		// The manager is unlocked in callback thread
+		if (idx >= 0) {
```

Comment:
Have you actually seen this being called with idk < 0? It is of course fine to test; but is this more like an assertion? Should it ever be normal that it happens? Is it an error that shouldn't go unnoticed?

Reply:
I have not seen this happen and yes I don't think it should happen ever either. Assertions are sanity checks, but the MinSizeRel build type turns assert() into an empty macro, so that cannot be used.

Adding an error trace would be an option I guess, but idk if PX4_ERR() works from header files (no MODULE_NAME).

Reply:
Yes, you are right; PX4_ERR is not good in header, just leave this as is